### PR TITLE
fix(gatsby): fix bundling image cdn tools

### PR DESCRIPTION
**Pull request by pieh**
## Description

Currently some IMAGE_CDN utils are failing to bundle correctly, mostly due to dynamic imports that remain unhandled. This adds various aliases and tricks to make it work after bundling

## Related Issues

[ch54853]

### CODE REVIEW

File: packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import sharp from "sharp"
```
Comment (pieh) [inline packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js:1]: not sure if importing `sharp` here is correct. `gatsby` doesn't depend on it, but this module should be alias for `gatsby-sharp`, so not sure now if context of imports here is `gatsby` or `gatsby-sharp`

Comment (wardpeet) [reply packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js:1]: Not 100% sure what you mean here. Gatsby-sharp is an async wrapper for requiring sharp

Comment (pieh) [reply packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js:1]: I mean this file is in `gatsby` which doesn't have `sharp` as dep, but is `gatsby-sharp` dep, so not sure if I can just import it here like that

File: packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/cache.ts [hunk 2]
```diff
@@ -2,11 +2,16 @@ import importFrom from "import-from"
 import type { GatsbyCache } from "gatsby"
 
 export function getCache(): GatsbyCache {
+  if (global.polyfill_remote_file_cache) {
```
[the remark this answers is not in the document]
Comment (pieh) [reply packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/cache.ts:5]: Yeah, I will standardise this

File: packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js (hunk 1 above)
Comment (wardpeet) [reply packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js:1]: Probably not in PNP world, what we can do is re-exports plain sharp form gatsby-sharp/sharp

**Review by LekoArts — DISMISSED** [review]

File: packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js (hunk 1 above)
Comment (pieh) [reply packages/gatsby/src/schema/graphql-engine/shims/gatsby-sharp.js]: Done in [gatsbyjs/gatsby@`74bc1ad` (#36534)](https://github.com/gatsbyjs/gatsby/pull/36534/commits/74bc1ad2cc1ff3469f5628e2fabd9a4c38d451e4)

**Pull request merged**